app/experimenter/experiments/tests/api/v5/test_mutations.py

```diff
@@ -633,3 +633,104 @@ def test_end_experiment_fails_with_nonlive_status(self):
+            status=NimbusExperiment.Status.DRAFT,
+            risk_revenue=False,
+        )
+        response = self.query(
+            UPDATE_EXPERIMENT_MUTATION,
+            variables={
+                "input": {
+                    "id": experiment.id,
+                    "riskRevenue": True,
+                    "changelogMessage": "test changelog message",
+                }
+            },
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+
+        experiment = NimbusExperiment.objects.get(id=experiment.id)
+        self.assertEqual(experiment.risk_revenue, True)
+
+    def test_risk_partner_related(self):
+        user_email = "user@example.com"
+        experiment = NimbusExperimentFactory.create(
+            status=NimbusExperiment.Status.DRAFT,
+            risk_partner_related=False,
+        )
+        response = self.query(
+            UPDATE_EXPERIMENT_MUTATION,
+            variables={
+                "input": {
+                    "id": experiment.id,
+                    "riskPartnerRelated": True,
+                    "changelogMessage": "test changelog message",
+                }
+            },
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+
+        experiment = NimbusExperiment.objects.get(id=experiment.id)
+        self.assertEqual(experiment.risk_partner_related, True)
+
+    def test_risk_fields_fails_with_nondraft_status(self):
+        user_email = "user@example.com"
+        experiment = NimbusExperimentFactory.create(
+            status=NimbusExperiment.Status.COMPLETE,
+            risk_brand=False,
+            risk_revenue=False,
+            risk_partner_related=False,
+        )
+        response = self.query(
+            UPDATE_EXPERIMENT_MUTATION,
+            variables={
+                "input": {
+                    "id": experiment.id,
+                    "riskBrand": True,
+                    "riskRevenue": True,
+                    "riskPartnerRelated": True,
+                    "changelogMessage": "test changelog message",
+                }
+            },
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+
+        content = json.loads(response.content)
+        result = content["data"]["updateExperiment"]
+        message = result["message"]["experiment"][0]
+        self.assertIn("risk_brand", message)
+        self.assertIn("risk_revenue", message)
+        self.assertIn("risk_partner_related", message)
+
+        experiment = NimbusExperiment.objects.get(id=experiment.id)
+        self.assertEqual(experiment.risk_brand, False)
+        self.assertEqual(experiment.risk_revenue, False)
+        self.assertEqual(experiment.risk_partner_related, False)
```

Comment:
Are these really necessary as separate tests?  We have tests that test the update functionality of the mutation, you could fold these fields in there?

And the last test is repeating other tests to make sure you can't edit at non draft status?  Maybe a duplicate test?

Reply:
Yeah, thanks for caling this out. Wasn't sure how thorough I should be with the non-draft updates and testing everything individually. Will adjust.